packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -1,324 +1,167 @@
-		this.state = {
-			numberOfColumns: MIN_COL_NUM,
-		};
-
-		Dimensions.addEventListener( 'change', this.onLayout );
-	}
-
-	componentDidMount() {
-		this.props.showInsertionPoint();
-	}
 
-	componentWillUnmount() {
-		this.props.hideInsertionPoint();
-		Dimensions.removeEventListener( 'change', this.onLayout );
-	}
-
-	calculateMinItemWidth( bottomSheetWidth ) {
-		const { paddingLeft, paddingRight } = styles.columnPadding;
-		return (
-			( bottomSheetWidth - 2 * ( paddingLeft + paddingRight ) ) /
-			MIN_COL_NUM
-		);
-	}
+import InserterSearchResults from './search-results';
+import { store as blockEditorStore } from '../../store';
 
-	calculateItemWidth() {
+function InserterMenu( {
+	onSelect,
+	onDismiss,
+	rootClientId,
+	clientId,
+	isAppender,
+	shouldReplaceBlock,
+	insertionIndex,
+} ) {
+	const {
+		showInsertionPoint,
+		hideInsertionPoint,
+		clearSelectedBlock,
+		insertBlock,
+		removeBlock,
+		resetBlocks,
+		insertDefaultBlock,
+	} = useDispatch( blockEditorStore );
+
+	const {
+		items,
+		destinationRootClientId,
+		getBlockOrder,
+		getBlockCount,
+		canInsertBlockType,
+	} = useSelect( ( select ) => {
 		const {
-			paddingLeft: itemPaddingLeft,
-			paddingRight: itemPaddingRight,
-		} = InserterButton.Styles.modalItem;
-		const { width: itemWidth } = InserterButton.Styles.modalIconWrapper;
-		return itemWidth + itemPaddingLeft + itemPaddingRight;
-	}
-
-	calculateColumnsProperties() {
-		const bottomSheetWidth = BottomSheet.getWidth();
-		const { paddingLeft, paddingRight } = styles.columnPadding;
-		const itemTotalWidth = this.calculateItemWidth();
-		const containerTotalWidth =
-			bottomSheetWidth - ( paddingLeft + paddingRight );
-		const numofColumns = Math.floor( containerTotalWidth / itemTotalWidth );
+			getInserterItems,
+			getBlockRootClientId,
+			getBlockSelectionEnd,
+			...selectBlockEditorStore
+		} = select( blockEditorStore );
 
-		if ( numofColumns < MIN_COL_NUM ) {
-			return {
-				numOfColumns: MIN_COL_NUM,
-				itemWidth: this.calculateMinItemWidth( bottomSheetWidth ),
-				maxWidth: containerTotalWidth / MIN_COL_NUM,
-			};
+		let targetRootClientId = rootClientId;
+		if ( ! targetRootClientId && ! clientId && ! isAppender ) {
+			const end = getBlockSelectionEnd();
+			if ( end ) {
+				targetRootClientId = getBlockRootClientId( end ) || undefined;
+			}
 		}
+
 		return {
-			numOfColumns: numofColumns,
-			maxWidth: containerTotalWidth / numofColumns,
+			items: getInserterItems( targetRootClientId ),
+			destinationRootClientId: targetRootClientId,
+			getBlockOrder: selectBlockEditorStore.getBlockOrder,
+			getBlockCount: selectBlockEditorStore.getBlockCount,
+			canInsertBlockType: selectBlockEditorStore.canInsertBlockType,
 		};
-	}
-
-	onClose() {
-		// if should replace but didn't insert any block
-		// re-insert default block
-		if ( this.props.shouldReplaceBlock ) {
-			this.props.insertDefaultBlock();
+	} );
+
+	const { getBlockType } = useSelect( ( select ) => select( blocksStore ) );
```

Comment:
What do you think about whether it is helpful to pass an `[]` as a second argument here? It looks like `useSelect` takes two arguments, and it obviously works fine omitting the second argument, but I was just curious if you had thoughts on this. It looks to me like it's a bit more common in the project to pass `[]`, but that is far from a consistent approach (definitely a number of calls with only one argument), so either seems fine from a project perspective. I'm more just asking to learn from your thoughts.

Reply:
Good catch. There's probably a performance gain to calling useSelect with an empty array. It's probably minimal in this specific case since the menu component is short lived and (at least in this current state) is not very dynamic. 
That does change a bit once search is added. I'll add the `[]` in then next change set. Thanks!